### ISSUE TITLE
Finetuning produces silence

### ISSUE THREAD

**Issue by hippotabek** [quickvc/QuickVC-VoiceConversion#7]
Hello! Thank you for sharing your work!

When I am finetuning the model from your checkpoint for a specific voice for some reason I am getting silence when generating audio post-finetuning. 

Have you faced issue like this? Or have any guess what can be the issue?

**Comment by quickvc** [comment]
It is total silence,[0,0,...,0]. or just a very small number? If it is small, multiplying it by 32768 may work.

**Comment by hippotabek** [comment]
it is some very small number and when i multiply by 32768 it is producing noise. is there something specific about architecture that would require large number of iterations for finetuning? btw the only changes i made was decrease batch size to 16 and remove batch sample because it was giving error. and of course i preprocessed my dataset.

**Comment by hippotabek** [comment]
and at zeroth iteration (with initial checkpoint weight) the produced audio is good. so the weights are loaded correctly

**Comment by quickvc** [comment]
What error happened when using batch sampler? If possible, please try to use batch sampler, it is important I think.
This is the discussion about the batch sampler: #https://github.com/OlaWod/FreeVC/issues/32 , sorry it is in Chinese, please translate it.

**Comment by hippotabek** [comment]
i fixed batch sample issue, but it didn't help. i checked what's going on by saving y and y_hat during training. y is being normalized by 32768 in the data_utils, and when i play it i cannot hear anything and have to multiply it by 32768. But when I do same for y_hat in the training loop, it's just giving noise.

**Comment by quickvc** [comment]
I think I don't change anything in the code. Could you please train from the checkpoint and check the torch.max and torch.min of y_mel, y_hat_mel in https://github.com/quickvc/QuickVC-VoiceConversion/blob/277118de9c81d1689e16be8a43408eda4223553d/train.py#L169
and:
`y_true_mel = mel_spectrogram_torch(
          y.squeeze(1), 
          hps.data.filter_length, 
          hps.data.n_mel_channels, 
          hps.data.sampling_rate, 
          hps.data.hop_length, 
          hps.data.win_length, 
          hps.data.mel_fmin, 
          hps.data.mel_fmax
      )`

**Comment by hippotabek** [comment]
`
----y_mel------
max mel -8.397790908813477
min mel -9.439289093017578
-----y_true_mel-----
max mel -8.397790908813477
min mel -9.439289093017578

----y_mel------
max mel -9.13985824584961
min mel -9.439289093017578
-----y_true_mel-----
max mel -9.13985824584961
min mel -9.439289093017578


----y_mel------
max mel -9.009064674377441
min mel -9.439289093017578
-----y_true_mel-----
max mel -9.009064674377441
min mel -9.439289093017578`

**Comment by hippotabek** [comment]
-----y_true_mel-----
max mel 1.3469260931015015
min mel -9.430488586425781
----y_mel------
max mel -8.761229515075684
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.5952036380767822
min mel -9.431591033935547
----y_mel------
max mel -9.09759521484375
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.4025744199752808
min mel -9.431845664978027
----y_mel------
max mel -8.707639694213867
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.5493190288543701
min mel -9.432464599609375
----y_mel------
max mel -8.717413902282715
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.4902350902557373
min mel -9.431824684143066
----y_mel------
max mel -9.063760757446289
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.3892061710357666
min mel -9.432854652404785
----y_mel------
max mel -9.088297843933105
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.293563961982727
min mel -9.432817459106445

**Comment by quickvc** [comment]
how about y_hat_mel?

**Comment by hippotabek** [comment]
`----y_mel------
max mel -9.088297843933105
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.293563961982727
min mel -9.432817459106445
-----y_hat_mel-----
max mel -6.933274269104004
min mel -9.439057350158691




----y_mel------
max mel -9.007279396057129
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.2343993186950684
min mel -9.432376861572266
-----y_hat_mel-----
max mel -7.390188694000244
min mel -9.439108848571777




----y_mel------
max mel -9.06143569946289
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.3563562631607056
min mel -9.432671546936035
-----y_hat_mel-----
max mel -6.566359519958496
min mel -9.439214706420898`

**Comment by quickvc** [comment]
How about mel?
`mel = spec_to_mel_torch(
          spec, 
          hps.data.filter_length, 
          hps.data.n_mel_channels, 
          hps.data.sampling_rate,
          hps.data.mel_fmin, 
          hps.data.mel_fmax)`

**Comment by hippotabek** [comment]
`----mel------
max mel -8.894758224487305
min mel -9.439289093017578
----y_mel------
max mel -9.009064674377441
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.2343993186950684
min mel -9.432307243347168
-----y_hat_mel-----
max mel -6.89412260055542
min mel -9.43864917755127




----mel------
max mel -8.728700637817383
min mel -9.439289093017578
----y_mel------
max mel -8.980888366699219
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.5070302486419678
min mel -9.43191909790039
-----y_hat_mel-----
max mel -6.438241958618164
min mel -9.43895435333252




----mel------
max mel -8.73178482055664
min mel -9.439289093017578
----y_mel------
max mel -9.0255126953125
min mel -9.439289093017578
-----y_true_mel-----
max mel 1.470967411994934
min mel -9.429850578308105
-----y_hat_mel-----
max mel -6.528789520263672
min mel -9.438985824584961`

**Comment by hippotabek** [comment]
also let me tell you why i am running this experiment. Maybe you would be able to provide an answer. 

FreeVC saves the emotional tone of the speech during conversion. For example, if the source speech is in excited tone, then the converted voice will also be excited. What I noticed with QuickVC is that the emotional tone gets lost. And I wanted to know whether it would preserve the emotion tone of the speaker it's been finetuned on. For example, if I finetune on target speaker whose speech is excited, would converted speech always be excited or not. 

Do you have any insights regarding this?

**Comment by quickvc** [comment]
The input of FreeVC is wavlm, but quickvc is hubert-soft. Most of information in Hubert-soft is content information.

I train the checkpoint in VCTK, this is the my output:

mel tensor(1.5599, device='cuda:0') tensor(-9.4222, device='cuda:0')
y_true_mel tensor(1.5599, device='cuda:0') tensor(-9.4222, device='cuda:0')
y_mel tensor(1.3359, device='cuda:0') tensor(-9.4222, device='cuda:0')
y_hat_mel tensor(1.1264, device='cuda:0', grad_fn=<MaxBackward1>) tensor(-9.4185, device='cuda:0', grad_fn=<MinBackward1>)
mel tensor(1.5671, device='cuda:0') tensor(-9.4223, device='cuda:0')
y_true_mel tensor(1.5671, device='cuda:0') tensor(-9.4223, device='cuda:0')
y_mel tensor(1.1883, device='cuda:0') tensor(-9.4209, device='cuda:0')
y_hat_mel tensor(1.1029, device='cuda:0', grad_fn=<MaxBackward1>) tensor(-9.4251, device='cuda:0', grad_fn=<MinBackward1>)
mel tensor(1.3939, device='cuda:0') tensor(-9.4221, device='cuda:0')
y_true_mel tensor(1.3939, device='cuda:0') tensor(-9.4221, device='cuda:0')
y_mel tensor(1.2324, device='cuda:0') tensor(-9.4218, device='cuda:0')
y_hat_mel tensor(1.0557, device='cuda:0', grad_fn=<MaxBackward1>) tensor(-9.4210, device='cuda:0', grad_fn=<MinBackward1>)

Did you change mel_processing.py?

**Comment by quickvc** [comment]
I guess there is something wrong with spec, how did you get the spec, by spectrogram_torch in mel_processing?

**Comment by quickvc** [comment]
> I guess there is something wrong with spec, how did you get the spec, by spectrogram_torch in mel_processing?

If possible, please try VITS's mel_processing.py, which must be true

**Comment by hippotabek** [comment]
-----y_true_mel-----
max mel 0.8283382654190063
min mel -9.409659385681152
min value is  tensor(-52.2115, device='cuda:0', grad_fn=<MinBackward1>)
max value is  tensor(55.1887, device='cuda:0', grad_fn=<MaxBackward1>)
-----y_hat_mel-----
max mel 3.9118852615356445
min mel -4.34147834777832
-----y_true_mel-----
max mel 1.5273504257202148
min mel -9.391952514648438
min value is  tensor(-8.4472, device='cuda:0', grad_fn=<MinBackward1>)
max value is  tensor(7.3443, device='cuda:0', grad_fn=<MaxBackward1>)
-----y_hat_mel-----
max mel 2.943667411804199
min mel -5.06384801864624


So values i was showing u before were values when i was finetuning on LJSpeech. Now i just checked on some VCTK speaker and it's giving sensible values during finetuning.

But now it's generation noise for some reason when I checked after 200 iterations just to test things.

I didn't touch any of the preprocessing you have used.

**Comment by hippotabek** [comment]
So after 3000 steps on VCTK, it's producing some very bad quality audio. here is a  [vctk finetune sample](https://drive.google.com/file/d/1AOYPlweHzd8bzypy-MUJqGzvEe5dge6L/view?usp=sharing). I would expect that quality wouldn't degrade

**Comment by guoyingying432** [comment]
mel(which is generated by spec) and y_true_mel(which is generated by real wav y) must be totally same, if they are not same. please check dataset and dataloader carefully

**Comment by hippotabek** [comment]
i found the issue. there is an issue with the function that reads wav files. for one dataset it was reading values in the range between -1 to 1, so already normalized, while for some values were not normalized yet.

Thank you very much for ur time and patience!
